blosc/schunk.c

```diff
@@ -293,41 +279,108 @@ blosc2_schunk* blosc2_schunk_copy(blosc2_schunk *schunk, blosc2_storage storage)
       BLOSC_TRACE_ERROR("Can not get `usermeta` from schunk");
       return NULL;
     }
-    if (blosc2_update_usermeta(new_schunk, usermeta, usermeta_len, *storage.cparams) < 0) {
+    if (blosc2_update_usermeta(new_schunk, usermeta, usermeta_len, cparams) < 0) {
       BLOSC_TRACE_ERROR("Can not update the `usermeta`.");
       return NULL;
     }
+    free(usermeta);
   }
   return new_schunk;
 }
 
 
 /* Open an existing super-chunk that is on-disk (no copy is made). */
-blosc2_schunk* blosc2_schunk_open(const blosc2_storage storage) {
-  if (storage.urlpath == NULL) {
-    BLOSC_TRACE_ERROR("You need to supply a storage.urlpath.");
+blosc2_schunk* blosc2_schunk_open(char* urlpath) {
```

Comment:
Should be `const char *`?

Reply:
Ok.  It *was* `const char *` before, but CLion was issuing warnings down in the inner code and this was my easiest fix.  I can revisit that again, although I don't think this is that important.

Reply:
Fixed.